builtin/logical/transit/path_certificates.go

```diff
@@ -0,0 +1,261 @@
+package transit
+
+import (
+	"context"
+	"crypto/x509"
+	"encoding/pem"
+	"errors"
+	"fmt"
+	"strings"
+
+	"github.com/openbao/openbao/sdk/v2/framework"
+	"github.com/openbao/openbao/sdk/v2/helper/errutil"
+	"github.com/openbao/openbao/sdk/v2/helper/keysutil"
+	"github.com/openbao/openbao/sdk/v2/logical"
+)
+
+func (b *backend) pathCreateCSR() *framework.Path {
+	return &framework.Path{
+		Pattern: "keys/" + framework.GenericNameRegex("name") + "/csr",
+		Fields: map[string]*framework.FieldSchema{
+			"name": {
+				Type:        framework.TypeString,
+				Required:    true,
+				Description: "Name of the key to create a CSR for.",
+			},
+			"version": {
+				Type:        framework.TypeInt,
+				Required:    false,
+				Description: "The version of the key to create a CSR for. If not set, the latest version, `latest`, will be used.",
+			},
+			"csr": {
+				Type:        framework.TypeString,
+				Required:    false,
+				Description: "PEM encoded CSR template to use. The information attributes will be used as a basis for the CSR with the key in transit. If not set, a default template will be used.",
+			},
+		},
+		Operations: map[logical.Operation]framework.OperationHandler{
+			logical.UpdateOperation: &framework.PathOperation{
+				Callback: b.pathCreateCSRWrite,
+				DisplayAttrs: &framework.DisplayAttributes{
+					OperationVerb: "write",
+				},
+			},
+		},
+		HelpSynopsis:    pathCreateCSRHelpSynopsis,
+		HelpDescription: pathCreateCSRHelpDescription,
+	}
+}
+
+func (b *backend) pathImportCertChain() *framework.Path {
+	return &framework.Path{
+		Pattern: "keys/" + framework.GenericNameRegex("name") + "/set_certificate",
```

Comment:
```suggestion
		Pattern: "keys/" + framework.GenericNameRegex("name") + "/set-certificate",
```

Based on https://developer.hashicorp.com/vault/api-docs/secret/transit#set-certificate-chain :-)